src/diffusers/image_processor.py

```diff
@@ -85,7 +85,7 @@ def numpy_to_pil(images: np.ndarray) -> PIL.Image.Image:
         """
         if images.ndim == 3:
             images = images[None, ...]
-        images = (images * 255).round().astype("uint8")
+        images = (images.clip(0, 1) * 255).round().astype("uint8")
```

Comment:
The normalize/unnormalize path handles clipping. But if the image does not go through the normalize path the range might need to be clipped still

Reply:
Hmm not a fan of performing clipping here. Why would the image not go through the normalization path? I wonder if the other utilities in the processor module need updates to facilitate that. 

Cc: @yiyixuxu.

Reply:
There's a flag for normalize that can be not set, and the muse vae doesn't set it. It feels nit-picky to me to not allow clipping here

Reply:
Hmm I think the problem is that we're changing the function here: 

```py
(np.array([-1, 1.1]) * 255).round().astype("uint8")  # array([ 1, 24], dtype=uint8) 
```

is not the same as:

```py
(np.array([-1, 1.1]).clip(0, 1) * 255).round().astype("uint8")   # array([  0, 255], dtype=uint8)  
```

There might be weird edge cases where people pass values that are out of range no?

Reply:
I think it's the same as if you pass an out of range value to the normalized path, you expect the out of range value to be set to the min or max pixel depending on which side it's on.

If you pass an out of range value w/out clipping, it overflows on the type cast which imo can't be the expected behavior if you pass an out of range value so I don't think there's edge cases where people would expect that behavior

```
In [7]: np.array([260]).astype(np.uint8)
Out[7]: array([4], dtype=uint8)
```

Reply:
I would feel safer to just leave as is here. IF it's a bug let's maybe correct it in a follow-up PR

Reply:
moved clip to pipelines